software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+end instrButtMouseUp
+
+on resoButtMouseUp
+   put "on resoButtMouseUP" into xX["write"];    devLog xX
+   -- adjList already called on every change
+   --   topoSort sMainShort,sTopoSort,sNotVisited,sNoteList
+   supplyDemand sMainShort,sScopeSketch,sResources,sSomethingChanged,sNotVisited
+   --   put summarizeResources(sMain,sScopeSketch,sResources,sNowDoc) into tRes
+   put summarizeResourcesHTML(sMain,sMainShort,sScopeSketch,sResources,sNowDoc) into tRes
+   --   answer tRes   
+   -- create or overwrite the instructions.html file
+   put sMain[sNowDoc][0]["filepath"] into tPath -- path for the *.how file
+   set the itemdelimiter to slash
+   put "resources.html" into item -1 of tPath
+   put tRes into URL ("file:" & tPath)
+   launch url ("file:" & tPath)
+end resoButtMouseUp
+
+on saveButtMouseUp
+   put "on saveButtMouseUP" into xX["write"];    devLog xX
+   if sMain[sNowDoc][0]["filepath"] is empty then
+      saveAsButtMouseUp
+   else
+      --      dispatch "cmdJsonFromArray" to stack "easyJSON" with sMain[sNowDoc]
+      --      put cmdJsonFromArray(sMain[sNowDoc]) into tResult
+      put jsonFromArray(sMain[sNowDoc]) into tResult
+      --      put the result into URL ("file:" & sMain[sNowDoc][0]["filepath"])
+      put tResult into URL ("file:" & sMain[sNowDoc][0]["filepath"])
+      put "saved" into sMain[sNowDoc][0]["saveStatus"]
+      set itemdelimiter to slash
+      put item -1 of sMain[sNowDoc][0]["filepath"] into tForShow
+      replace ".how" with "" in tForShow
+      put tForShow into field "filenameLabel" stack "HowstrBeta"
+   end if
+end saveButtMouseUp
+
+on loadButtMouseUp
+   -- on the main stack; as in load a Howstr *.how file
+   put "on loadButtMouseUp" into xX["write"];    devLog xX
+   if sMain[sNowDoc][0]["saveStatus"] is "unsaved" then
+      answer "Load new without saving current?" with "Close current" or "Cancel"
+      if it is "cancel" then exit loadButtMouseUp
+   end if
+   put specialFolderPath("documents") into tDocsPath
+   answer folder "Select project folder"  with tDocsPath
+   if the result is not "cancel" then
+      put it into tReq["newProjectFolder"]
+      put loadNewProject(tReq) into tFileData
+      if not (tFileData["data"] begins with "Error") then
+         put empty into sMain
+         put arrayFromJson(tFileData["data"]) into sMain[1]
+         put 1 into sNowDoc
+         -- filepath
+         put tFileData["fullPath"] into sMain[sNowDoc][0]["filepath"] -- absolute path to *.how file
+         set itemdelimiter to slash
+         put item -1 of tFileData["fullPath"] into tForShow
+         replace ".how" with "" in tForShow
+         put tForShow into field "filenameLabel" stack "HowstrBeta"
+         refreshDraw
+         refreshDisplay
+      end if
+   end if
+   -- this will overwrite the "unsaved" that the build handler flags
+   -- since in this case there aren't any unsaved changes yet
+   put "opened" into sMain[sNowDoc][0]["saveStatus"]
+   
```

Comment:
I don't understand this comment. Load doesn't create any files, it just reads the JSON into an array and replaces whatever was in the working array before, then starts the graph over again.